### PR TITLE
Added RSQL feature, also corrected validation error responses

### PR DESCRIPTION
**Pull request by RussellSk**

### CODE REVIEW

File: sandbox-backend/src/main/java/com/exadel/team2/sandbox/configuration/ErrorHandlingControllerAdvice.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+import javax.validation.ConstraintViolation;
+import javax.validation.ConstraintViolationException;
+
+@ControllerAdvice
+class ErrorHandlingControllerAdvice {
```
Comment (akarzhaneuski) [inline sandbox-backend/src/main/java/com/exadel/team2/sandbox/configuration/ErrorHandlingControllerAdvice.java:18]: Good approach to decorate exceptions! Just small note than class and methods should be public in ```@ControllerAdvice ```, package private might fail in some cases

**Review by akarzhaneuski — CHANGES_REQUESTED** [review]

**Review by ikantsevich — APPROVED** [review]

File: sandbox-backend/src/main/java/com/exadel/team2/sandbox/configuration/ErrorHandlingControllerAdvice.java (hunk 1 above)
Comment (RussellSk) [reply sandbox-backend/src/main/java/com/exadel/team2/sandbox/configuration/ErrorHandlingControllerAdvice.java:18]: Fixed access modifier. Thank you for advice

File: sandbox-web/src/main/java/com/exadel/team2/sandbox/controller/EmployeeController.java [hunk 2]
```diff
@@ -25,10 +33,17 @@ public ResponseEmployeeDto getEmployeeById(@PathVariable Long id) {
...
 
     @GetMapping
-    public Page<ResponseEmployeeDto> getEmployees(
+    public Page<ResponseEmployeeDto> getEmployeesRsql(
+            @RequestParam(value = "search") String search,
             @RequestParam(defaultValue = "0", name = "page") Integer page,
             @RequestParam(defaultValue = "15", name = "itemsPerPage") Integer itemsPerPage) {
-        return employeeService.getAllPageable(PageRequest.of(page, itemsPerPage));
+        try {
+            Node rootNode = new RSQLParser().parse(search);
```
[the remark this answers is not in the document]
Comment (ikantsevich) [reply sandbox-web/src/main/java/com/exadel/team2/sandbox/controller/EmployeeController.java]: Yes

[the remark this answers is not in the document]
Comment (RussellSk) [reply sandbox-web/src/main/java/com/exadel/team2/sandbox/controller/EmployeeController.java]: Thank you!

[the remark this answers is not in the document]
Comment (RussellSk) [reply sandbox-web/src/main/java/com/exadel/team2/sandbox/controller/EmployeeController.java]: Done. Moved RSQL logic from Controller to Service

**Review by akarzhaneuski — APPROVED** [review]

**Pull request merged**